### DISCUSSION TITLE
Langfuse session ID when using Chainlit

### DISCUSSION THREAD

**Discussion by tituslhy** [langfuse/langfuse#1449] [Support]
Hello everyone,

I tried using out the new langfuse llamaindex capabilities in my chainlit app. It works great except that the set_trace_params doesn't seem to be working well - my traces are all over the place instead of being tagged to a session ID.

Here's my sample code
```
import chainlit as cl
from llama_index.core.settings import Settings
from llama_index.core.callbacks import CallbackManager
from langfuse.llama_index import LlamaIndexCallbackHandler
### Other imports ###

# Load langfuse environ vars
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())

langfuse_callback_handler = LlamaIndexCallbackHandler()

@cl.on_chat_start
def start_chat():
    langfuse_callback_handler.set_trace_params(
        session_id="test"
    )
    
    #### Codes to instantiate llm, etc. ###
    
    Settings.callback_manager = CallbackManager([
        cl.LlamaIndexCallbackHandler(),
        langfuse_callback_handler
    ])

@cl.on_message
def on_message(message):
    #### Codes to tell chainlit how to handle user messages. ###
    pass

@cl.on_chat_end
def on_chat_end():
    langfuse_callback_handler.flush()
    #### Other codes ###
    
```
Has anyone tried langfuse and llamaindex on chainlit?

**Comment by marcklingen** [comment]
unfortunately not an expert on chainlit, hope someone else can help you out here

**Reply by DhavalThkkar** [reply]
@marcklingen I think Chainlit is definitely worth pursuing for traceability. They have their own solution called Literal AI with which everything is very closely tied to. I feel like this integration can pull in a lot of users to utilise langfuse to a very great extent. 

The only place where I think the integration of Langfuse with Chainlit would be problematic is how they handle user feedback. If that gets resolved, Langfuse should be good to go!

**Comment by ghost** [comment]
Hi @tituslhy, did you manage to find a solution to this?

**Reply by tituslhy** [reply]
Not at all I'm afraid. I just changed to use a different traceability tool because of business exigencies.
